R/xts.R

```diff
@@ -15,11 +15,14 @@ as.data.table.xts = function(x, keep.rownames = TRUE, key=NULL, ...) {
   r[]
 }
 
-as.xts.data.table = function(x, ...) {
+as.xts.data.table = function(x, numeric.only = TRUE, ...) {
   stopifnot(requireNamespace("xts"), !missing(x), is.data.table(x))
   if (!xts::is.timeBased(x[[1L]])) stopf("data.table must have a time based column in first position, use `setcolorder` function to change the order, or see ?timeBased for supported types")
-  colsNumeric = vapply_1b(x, is.numeric)[-1L] # exclude first col, xts index
-  if (!all(colsNumeric)) warningf("Following columns are not numeric and will be omitted: %s", brackify(names(colsNumeric)[!colsNumeric]))
-  r = setDF(x[, .SD, .SDcols = names(colsNumeric)[colsNumeric]])
-  return(xts::as.xts(r, order.by = if ("IDate" %chin% class(x[[1L]])) as.Date(x[[1L]]) else x[[1L]]))
+  r <- x[, -1L]# exclude first col, xts index
```

Comment:
sorry for the late review. would it be better to `shallow()`-copy `x` and then delete the first column?